### ISSUE TITLE
Choosing "raw" should be the only required condition, in an auto-apply preset, if the preset should apply to all raw files

### ISSUE THREAD

**Issue by OK-One** [darktable-org/darktable#9344]
**Is your feature request related to a problem? Please describe.**

![image](https://user-images.githubusercontent.com/71924830/123523522-ff933d00-d6bb-11eb-8c50-0054b7d839c1.png)

Why do I need to also select "HDR", "monochrome", and "colour" as additional conditions for an auto apply preset, in the sharpen module, when the only condition I needed was "raw". 

If I select raw as the only condition, the auto apply does not work. I would expect that raw should be the only condition I need to enable, without having to select the other conditions.  I've checked the entire manual for version 3.4, and nowhere does it say that "HDR", "monochrome" and "colour" are additional prerequisite conditions, which are also needed to make the "raw" condition work.

In programming terms, selecting raw or any other condition should be a boolean "OR".

"Monochrome" and "Colour" are mutually exclusive, i.e these conditions are therefore "OR", so anyone looking at the menu would expect that the other conditions like "Raw" are also "OR" conditions.

These are my thoughts, and I hope I can find some explanation for why I am not able to choose only one condition "raw".

**Describe the solution you'd like**

I hope someone can provide clarity on why I have to know some additional info, like enabling "HDR", "Monochrome", and "colour", as additional conditions to "raw", when all I should need to select "raw", in order to auto apply a preset to ALL raw files.

And this additional "knowledge" is not included in the manual., so my choosing only "raw" cannot be said to be user error.

If this is a convention, its not one that is obvious, and maybe needs to be included in the manual, until the usual behaviour that most people might expect to work, ie choosing only raw as the only condition (when you want it to apply to ALL raw files), has been revised/corrected in darktable., in a future version.

**Alternatives**

**Additional context**

**Comment by elstoc** [comment]
Clarification was provided on your other issue. I have raised an issue against the user manual (https://github.com/darktable-org/dtdocs/issues/289) and once 3.6 is out of the way I will update the manual.

**Comment by OK-One** [comment]
Good enough. Thanks. I appreciate that this will help other users, with the manual revised. 

Thank you.

**Comment by wpferguson** [comment]
> I hope someone can provide clarity on why I have to know some additional info, like enabling "HDR", "Monochrome", and "colour", as additional conditions to "raw", when all I should need to select "raw", in order to auto apply a preset to ALL raw files.

I will try and provide some clarity...

Let's say for the sake of argument that your proposal got accepted and the preset choices were re-coded.  

* Every module using presets is now broken and has to be fixed.  
* All of the supplied presets have to be re-coded and tested.  
* Every edit ever done in darktable, using a preset, is now broken.  
* New code needs to be written to try and address every preset ever thought of and used, and transition it to the new preset code. 

If we try and and satisfy everyone's opinion about how a module should work we'd spend all of our time working on one module and never get anything done, because everyone has an opinion and they're all different.

**Comment by OK-One** [comment]
> > I hope someone can provide clarity on why I have to know some additional info, like enabling "HDR", "Monochrome", and "colour", as additional conditions to "raw", when all I should need to select "raw", in order to auto apply a preset to ALL raw files.
> 
> I will try and provide some clarity...
> 
> Let's say for the sake of argument that your proposal got accepted and the preset choices were re-coded.
> 
> * Every module using presets is now broken and has to be fixed.
> * All of the supplied presets have to be re-coded and tested.
> * Every edit ever done in darktable, using a preset, is now broken.
> * New code needs to be written to try and address every preset ever thought of and used, and transition it to the new preset code.
> 
> If we try and and satisfy everyone's opinion about how a module should work we'd spend all of our time working on one module and never get anything done, because everyone has an opinion and they're all different.

Hi, 

To be clear, 1st thanks for your reply. I appreciate this.

You make a good point. There is the legacy of previous users, and their assets, which rely on the current behavior of the preset auto-apply logic.

What brought me to this whole sequence of inquests, was an attempt to use the features, based on the information available to me, included what's available in the manual.

Having used all the information available to me, I still was not able to achieve an objective because, I needed some additional information about using these presets, which was kindly revealed to me on one of my previous enquiries.

One of the darktable contributor, has kindly accepted to initiate a change in the manual, to include this missing item of knowledge required to use the preset system, based on the current logic.

I definitely appreciate that because of the legacy, it would need much more thought and effort to make any changes to the logic.

I am 100% satisfied, that with the change in the manual, anyone else will be able to achieve what I had wanted to do, because the missing information, that I had no way of knowing about, and which led me to make wrong assumptions about the way the preferences in darktable worked, will now be available to everyone.

With this alternative resolution, I think we can close the issue, as we all have a suitable way of dealing with the issue., a change to the information included in the manual. 

I am satisfied, and happy that I raised the issue, cos it led me to discover, something I could never have known, any other way. So the whole episode is all positive. (I had to raise three issues before I got to the bottom of this - pretty persistent). And I thank everyone who has been kind enough to contribute to a solution.
